File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_035)_Minimum_Group_flips_to_make_same.cpp
#include <iostream>
using namespace std;

int minGroupFlipSame(int arr[], int n)
{
    for (int i = 1; i < n; i++)
    {
        if (arr[i] != arr[i - 1])
        {
            if (arr[i] != arr[0])
                cout << "From " << i << " to ";
            else
                cout << i - 1 << endl;
        }
    }
    if (arr[n - 1] != arr[0])
        cout << n - 1 << endl;
}

int main()
{
    int arr[] = {1, 1, 0, 0, 0, 1, 0, 0, 1}, n = 9;
    minGroupFlipSame(arr, n);
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_035)_Minimum_Group_flips_to_make_same.dart
import 'dart:io';

void minGroupFlipSame(List<int> arr) {
  for (int i = 1; i < arr.length; i++) {
    if (arr[i] != arr[i - 1]) {
      if (arr[i] != arr[0])
        stdout.write('From $i to ');
      else
        stdout.writeln(i - 1);
    }
  }

  if (arr[arr.length - 1] != arr[0]) stdout.writeln(arr.length - 1);
}

void main() {
  var arr = [1, 1, 0, 0, 0, 0, 0, 1, 1, 0];
  minGroupFlipSame(arr);
}
